low-level/framework/utils/salt_util.py

```diff
@@ -13,88 +13,114 @@
+            if self.pillar_info:
+                cluster_type = \
+                    self.pillar_info[self.node_id][self.CLUSTER_KEY][self.CLUSTER_TYPE_KEY]
+                if cluster_type is not None and cluster_type.lower() == 'single':
+                    _is_single_node = True
             else:
-                if result[0].decode().rstrip('\n') != "single":
-                    is_single_node = False
-        except Exception as e:
-            logger.warning(f"Cluster type read failed, assuming single node setup : {e}")
-        return is_single_node
+                logger.warning(f'sspl_constants, Fail to read cluster type with \
+                               this error: {_err}, Assuming single node')
+        except Exception as err:
+            logger.warning(f'sspl_constants, Fail to read cluster type with \
+                               this error: {err}, Assuming single node')
+        logger.info(f'sspl_constants, Node type: {_is_single_node}')
+        return _is_single_node
 
-    def get_consul_vip(self, node_name):
+    def get_consul_vip(self):
         """
         Returns IP used to connect to Consul
         """
-        is_env_vm = self.utility.is_env_vm()
-        is_single_node = self.is_single_node()
+        _is_env_vm = self.utility.is_env_vm()
+        _is_single_node = self._is_single_node
         # Initialize to localhost
-        consul_vip = "127.0.0.1"
-        # Get vip if not a vm or single node, default to localhost otherwise
-        if not is_env_vm and not is_single_node:
-            # Read vip from cluster.sls
-            try:
-                SALT_CMD = f"sudo salt-call pillar.get cluster:{node_name}:network:data_nw:roaming_ip --output=newline_values_only"
-                subout = subprocess.Popen(SALT_CMD, shell=True, stdout=subprocess.PIPE, stderr=subprocess.PIPE)
-                result = subout.stdout.readlines()
-                if result == [] or result == "":
-                    logger.warning("Cluster VIP fetch failed, using localhost to connect to Consul.")
-                else:
-                    consul_vip = result[0].decode().rstrip('\n')
-            except Exception as e:
-                logger.warning(f"Cluster VIP read failed, using localhost to connect to Consul : {e}")
-        return consul_vip
+        _consul_vip = "127.0.0.1"
+        # Get vip if not a vm or single node, default to localhost otherwise if not _is_env_vm and not _is_single_node:
+        try:
+            if self.pillar_info is not None:
+                data_store_config = \
+                    self.pillar_info[self.node_id][self.SSPL_KEY][self.DATASTORE_KEY]
+                if data_store_config is not None:
+                    _consul_vip = data_store_config['consul_host']
+            else:
+                logger.warning(f'sspl_constants, Fail to read consul VIP with \
+                                 this error: {_err}, Assuming localhost: \
```

Comment:
Will there be any error info in _err here since we will not try if pillar_info is none

Reply:
Fixed this problem now.